### PR TITLE
Add a CustomConstantAttribute test (#49748)

### PR DESCRIPTION
**Pull request by SkiFoD**
There is a test the custom constant attribute assigns the default value to Type.Missing parameter.

### CODE REVIEW

**Comment by MichalStrehovsky** [comment]
> runtime (Libraries Test Run release mono Linux x64 Debug) Failing after 32m

Could you add a `[SkipOnMono("https://github.com/dotnet/runtime/issues/49806")]` attribute on the new test? Apparently this doesn't work with Mono.

File: src/libraries/System.Runtime/tests/System/DelegateTests.cs [hunk 1]
```diff
@@ -102,6 +102,28 @@ public static void DynamicInvoke()
...
         }
 
+        private class SomeCustomConstantAttribute : CustomConstantAttribute
+        {
+            private const string DEFAULT_VALUE = "SomeValue";
+
+            public static void Do(object o)
+            {
+                //Check the default value is assigned to the parameter as expected
+                Assert.Equal(DEFAULT_VALUE, o);
```
Comment (MichalStrehovsky) [inline src/libraries/System.Runtime/tests/System/DelegateTests.cs:112]: Nit: Maybe make this `public static object Do(object o) => o;` and put the assert in the test method itself. I think we tend to put asserts in the `[Fact]` methods themselves.

**Review by MichalStrehovsky — COMMENTED** [review]
Looks good otherwise! Thank you for looking into this!

File: src/libraries/System.Runtime/tests/System/DelegateTests.cs (hunk 1 above)
Comment (SkiFoD) [reply src/libraries/System.Runtime/tests/System/DelegateTests.cs:112]: Looks like a good idea to me. I'm going to change it.

**Review by MichalStrehovsky — APPROVED** [review]
Thank you!

The Android failures look unrelated. Merging.

**Pull request merged**